### ISSUE TITLE
Adopting SPDX-License-Identifier

### ISSUE THREAD

**Issue by hosiet** [flameshot-org/flameshot#1374]
TL;DR: Let's go with https://spdx.dev/ids/ instead of inserting lengthy copyright claiming header within each source file.

* * *

Details:

I was inspired by a previous discussion thread in flameshot's slack workspace, which points to https://github.com/flameshot-org/flameshot/pull/1131#discussion_r515650195 . I believe we have an easier and more canonical solution, which is to use SPDX identifier. This only requires inserting one line of machine-readable comment (or sometimes 2~3 lines, if copyright holder is to be added as well) instead of tens of lines of repeated paragraphs. It's being used by multiple projects, including the Linux kernel (see https://lwn.net/Articles/739183/ ).

One of my favourite upstream project (fcitx5) is using it and I'd take it as a straightforward example:

https://github.com/fcitx/fcitx5/blob/39c0a95b26c350a71c5deff4492c3589ec3725fa/src/server/errorhandler.cpp#L1-L13

```cpp
/*
 * SPDX-FileCopyrightText: 2002-2005 Yuking <yuking_net@sohu.com>
 * SPDX-FileCopyrightText: 2010-2015 CSSlayer <wengxt@gmail.com>
 *
 * SPDX-License-Identifier: LGPL-2.1-or-later
 *
 */

#include "config.h"

#include <fcntl.h>
#include <signal.h>
#include <string.h>
```

The literal description about its usage can be found at https://spdx.dev/ids/#how .

**Comment by borgmanJeremy** [comment]
Yeah this seems like a fine idea. If we want to do this lets hit all the files in one merge.
